rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/topic_cache.hpp

```diff
@@ -102,23 +128,34 @@ class TopicCache
     return participant_to_topics_;
   }
 
+  /**
+   * @return a map of topic name to a vector of GUID_t, type name and qos profile tuple.
+   */
+  const TopicNameToTopicData & getTopicNameToTopicData() const
+  {
+    return topic_name_to_topic_data_;
+  }
+
   /**
    * Add a topic based on discovery.
    *
    * @param rtpsParticipantKey
    * @param topic_name
    * @param type_name
+   * @param dds_qos the dds qos policy of the participant
    * @return true if a change has been recorded
    */
+  template<class T>
```

Comment:
`dds_qos` can be of type `WriterQos` or `ReaderQos`, but the function implementation remains the same regardless. `dds_qos_to_rmw_qos` (line 170) also takes in a template variable for `dds_qos`.

Reply:
That's fine, but consider `static_assert`ing `T` is a supported type.